Additional SELinux fixes, addition of NGINX catch-all and fix of RHEL Python bootstrap halt

thunderysteak opened the pull request:
Hey, I found out that the Vagrant box that I used for testing had SELinux set in permissive mode, so all SELinux policies were ignored. 

https://github.com/geerlingguy/packer-boxes/blob/master/rockylinux8/http/ks.cfg#L8

I spun up a bare VM with everything at default and tried to fix it.

This fixes NGINX's proxying and executing rbenv binaries as systemd service. 

I also have fixed a situation where Ansible halts during the bootstrapping task, as `yum update` only updates packages (and doesn't refresh the cache like with apt) and halts because it requires user input, plus renamed Amazon Linux to general RHEL.

I've added an option for nginx_catch_all where NGINX will display Mastodon for any IP/Domain for Reverse Proxy or Vagrant DHCP scenarios. 

Fixed scenario where letsencrypt was still executing even when the disable_letsencrypt variable was set to true

And last added .DS_Store to gitignore as its just MacOS clutter and fixed a typo in a comment.

moritzheiber commented:
@thunderysteak Can you re-base this on top of `main`? The tests are running for RHEL as well now; I'd love to get them to a point where we can trust them :slightly_smiling_face:

moritzheiber commented:
@thunderysteak You can ignore the test failure on Jammy, that's an incompatibility between Ruby's OpenSSL extension and Jammy's version of OpenSSL (1 vs. 3).

However, there are two tests for the RHEL variant, the PostgreSQL access (not sure why this one is failing) and the `nginx` configuration. I'm not familiar with RHEL's setup for `nginx` enabled configurations .. is `default.d` the place where `conf.d` sites are getting linked into to enable them?

For reference, the tests are run from `goss.yaml`, this is the command to verify the PostgreSQL access:

```console
PGPASSWORD=CHANGEME psql -d mastodon_instance -h 127.0.0.1 -U mastodon -c 'CREATE TABLE test (v varchar(20)); DROP TABLE test;'
```
(line 84 of the `goss.yaml`)

thunderysteak commented:
Hi @moritzheiber 

For RHEL, NGINX automatically grabs any configuration from `conf.d` and doesn't require simlinking as on Debian

This is an example from `/etc/nginx/nginx.conf` which is unmodified and default configuration. This basically includes anything from the `conf.d` directory like it was the main `nginx.conf` file.

```    
    # Load modular configuration files from the /etc/nginx/conf.d directory.
    # See http://nginx.org/en/docs/ngx_core_module.html#include
    # for more information.
    include /etc/nginx/conf.d/*.conf;
```

Regarding Postgres checks failing, its due of Postgres using IDENT-based authentication by default and this will never allow you to login via -U and -W options, additionally there's no ident provider installed.

Modifying `/var/lib/pgsql/data/pg_hba.conf` from the defaults:

```
# TYPE  DATABASE        USER            ADDRESS                 METHOD

# "local" is for Unix domain socket connections only
local   all             all                                     peer
# IPv4 local connections:
host    all             all             127.0.0.1/32            ident
# IPv6 local connections:
host    all             all             ::1/128                 ident
# Allow replication connections from localhost, by a user with the
# replication privilege.
local   replication     all                                     peer
host    replication     all             127.0.0.1/32            ident
host    replication     all             ::1/128                 ident
```
To this where we changed ident to md5
```
# TYPE  DATABASE        USER            ADDRESS                 METHOD

# "local" is for Unix domain socket connections only
local   all             all                                     peer
# IPv4 local connections:
host    all             all             127.0.0.1/32            ident
# IPv6 local connections:
host    all             all             ::1/128                 ident
# Allow replication connections from localhost, by a user with the
# replication privilege.
local   replication     all                                     peer
host    replication     all             127.0.0.1/32            ident
host    replication     all             ::1/128                 ident
```
Now permits logins and the test should now run fine:
```
[thunderysteak@mastodon ~]$ PGPASSWORD=CHANGEME psql -d mastodon_instance -h 127.0.0.1 -U mastodon -c 'CREATE TABLE test (v varchar(20)); DROP TABLE test;'
psql: FATAL:  Ident authentication failed for user "thunderysteak"
[thunderysteak@mastodon ~]$ sudo nano /var/lib/pgsql/data/pg_hba.conf
[thunderysteak@mastodon ~]$ sudo systemctl restart postgresql
[thunderysteak@mastodon ~]$ PGPASSWORD=CHANGEME psql -d mastodon_instance -h 127.0.0.1 -U mastodon -c 'CREATE TABLE test (v varchar(20)); DROP TABLE test;'
DROP TABLE
```

moritzheiber commented:
re: `nginx`: Great! Even better .. one file less to change.
re: PostgreSQL: I'm not particularly bound towards either method of authentication as long as it's easy for folks to connect to the database and/or it's possible right from the start. Is that the case without further modifications? Or should we change the `pg_hba.conf` in the way you suggested?

thunderysteak commented:
I was wrong about ident service missing for postgres and forgot to replace ident with md5 in my example.

Via CLI, users can easily modify or do edits to the database as long as they're logged in as the `mastodon` user and the -U flag is not used. This is the default behavior and shouldn't be changed just for tests. Ideally we just want some shell commands in Vagrantfile that replaces ident with md5, as Vagrant is purely for testing.

```
[thunderysteak@mastodon ~]$ sudo su - mastodon
Last login: Fri May 13 10:09:21 EDT 2022 on pts/0
[mastodon@mastodon ~]$ PGPASSWORD=CHANGEME psql -d mastodon_instance -h 127.0.0.1 -c 'CREATE TABLE test (v varchar(20)); DROP TABLE test;'
DROP TABLE
```

Just updated goss.yml and removed the default.d checks and added shell commands to vagrantfile to just replace ident with md5 using sed and reboot the service to make the tests happy

moritzheiber commented:
As long as the Ruby processes have access to the database I'm fine with it :slightly_smiling_face:

moritzheiber commented:
Thanks!
